src/prefect/agent.py

```diff
@@ -94,31 +91,31 @@ async def get_and_submit_flow_runs(self) -> List[FlowRun]:
             seconds=self.prefetch_seconds or PREFECT_AGENT_PREFETCH_SECONDS.value()
         )
 
-        # Use the work queue id or load one from the name
-        work_queue_id = self.work_queue_id or await self.work_queue_id_from_name()
-        if not work_queue_id:
-            return []
+        submittable_runs = []
 
-        try:
-            submittable_runs = await self.client.get_runs_in_work_queue(
-                id=work_queue_id, limit=10, scheduled_before=before
-            )
-        except httpx.HTTPStatusError as exc:
-            if exc.response.status_code == status.HTTP_404_NOT_FOUND:
-                raise ValueError(
-                    f"No work queue found with id '{work_queue_id}'"
-                ) from None
-            else:
-                raise
+        # load runs from each work queue
+        for work_queue in await self.get_work_queues():
```

Comment:
The previous overhead when there were no runs to submit is a bit moot since it has nothing better to do. When there are runs to submit is when we need to be fast.

I'm totally fine with leaving this as is for now. We can add caching in the future if necessary. I just wanted to make sure it was on our minds.

Reply:
I think the prefetch behavior should mitigate the need to be fast when there are runs -- we're loading them into the agent with up to 10 seconds of buffer time. I thought you were more concerned with the overhead on the server of the additional calls. In either case, we can examine and optimize in the future but I'm not overly concerned.

Reply:
Fair! 👍

Reply:
@madkinsz I added a 30 second cache